storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -166,6 +166,10 @@ public Cache<Uuid, Entry> internalCache() {
         return internalCache;
     }
 
+    public void remove(Uuid key) {
```

Comment:
please acquire a read lock here so that it doesn't conflict with cache closing. we use lock here to ensure that no other thread is accessing the cache when it is closing.

Reply:
Should we take read/write lock? Read lock can be held by multiple threads so took the write lock instead. Also, added one method to invalidate a collection of keys.

Reply:
Right, we should take write-lock here as we are mutating the structure.

Reply:
Nope. Write lock is not needed here folks. This is because the cache is thread-safe i.e. the caffeine interface internally ensures that multiple threads can read/write to it at the same time. 

We use a lock here to ensure that "close" mention of the RemoteIndeCache can execute in isolation i.e. no other method is called when this cache is being closed. That is why a read lock will suffice. 

Is this makes sense to you, I will create a JIRa and fix it. ok @satishd @kamalcph ?

Reply:
Sure, that makes sense.

Reply:
I'll fix this one with KAFKA-15290. Thanks for following up!

Reply:
Thank you Kamal and Satish!